pkg/service/agentservice.go

```diff
@@ -116,64 +103,55 @@ func (s *AgentService) ServeHTTP(writer http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	s.HandleConnection(r.Context(), conn)
+	s.HandleConnection(r, conn)
 }
 
-func NewAgentHandler(agentServer rpc.AgentInternalServer, roomTopic, publisherTopic string) *AgentHandler {
-	return &AgentHandler{
-		agentServer:      agentServer,
-		roomTopic:        roomTopic,
-		publisherTopic:   publisherTopic,
-		availability:     make(map[string]chan *availability),
-		unregistered:     make(map[*websocket.Conn]*worker),
-		roomWorkers:      make(map[string]*worker),
-		publisherWorkers: make(map[string]*worker),
+func (s *AgentService) HandleConnection(r *http.Request, conn *websocket.Conn) {
+	var protocol agent.WorkerProtocolVersion
+	if pv, err := strconv.Atoi(r.Form.Get("protocol")); err == nil {
```

Comment:
`request.Form` is not populated unless explicitly parsed. better to use [FormValue](https://pkg.go.dev/net/http#Request.FormValue)